File: front/apps/web-ele/src/api/core/department.ts
import { requestClient } from '#/api/request';

export interface Department {
  id: number;
  name: string;
  code: string;
  description: string;
  status: string;
  parent_id?: number;
  created_at: string;
  updated_at: string;
  deleted_at?: string;
}

export interface DepartmentListResponse {
  total: number;
  departments: Department[];
}

/**
 * 获取部门列表
 */
export async function getDepartmentListApi(params?: {
  status?: string;
  page?: number;
  page_size?: number;
}): Promise<DepartmentListResponse> {
  const response = await requestClient.get<DepartmentListResponse>('/department', { params });
  return response.data;
}

/**
 * 获取部门详情
 */
export async function getDepartmentDetailApi(id: number): Promise<Department> {
  const response = await requestClient.get<Department>(`/department/${id}`);
  return response.data;
}

/**
 * 创建部门
 */
export async function createDepartmentApi(data: {
  name: string;
  code?: string;
  description?: string;
  parent_id?: number;
}): Promise<Department> {
  const response = await requestClient.post<Department>('/department', data);
  return response.data;
}

/**
 * 更新部门
 */
export async function updateDepartmentApi(
  id: number,
  data: {
    name?: string;
    code?: string;
    description?: string;
    parent_id?: number;
  },
): Promise<Department> {
  const response = await requestClient.put<Department>(`/department/${id}`, data);
  return response.data;
}

/**
 * 停用部门
 */
export async function deactivateDepartmentApi(id: number): Promise<{ message: string }> {
  const response = await requestClient.put<{ message: string }>(
    `/department/${id}/deactivate`,
  );
  return response.data;
}

/**
 * 激活部门
 */
export async function activateDepartmentApi(id: number): Promise<{ message: string }> {
  const response = await requestClient.put<{ message: string }>(`/department/${id}/activate`);
  return response.data;
}

File: back/internal/user/domain/department.go
package domain

import (
	"time"
	"gorm.io/gorm"
)

// 部门状态常量
const (
	DepartmentStatusActive   = "active"   // 激活
	DepartmentStatusInactive = "inactive" // 停用
)

// Department 部门聚合根
type Department struct {
	ID          uint           `gorm:"primaryKey" json:"id"`
	Name        string         `gorm:"size:100;not null" json:"name"`
	Code        string         `gorm:"size:50;uniqueIndex" json:"code"`
	Description string         `gorm:"size:500" json:"description"`
	Status      string         `gorm:"size:20;default:active" json:"status"`
	ParentID    *uint          `gorm:"index" json:"parentId"`
	CreatedAt   time.Time      `gorm:"autoCreateTime" json:"createdAt"`
	UpdatedAt   time.Time      `gorm:"autoUpdateTime" json:"updatedAt"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

// TableName 表名
func (Department) TableName() string {
	return "departments"
}

// Validate 验证部门数据
func (d *Department) Validate() error {
	if d.Name == "" {
		return ErrDepartmentNameEmpty
	}

	if len(d.Name) < 2 || len(d.Name) > 100 {
		return ErrDepartmentNameInvalid
	}

	if d.Code != "" && (len(d.Code) < 2 || len(d.Code) > 50) {
		return ErrDepartmentCodeInvalid
	}

	return nil
}

// Deactivate 停用部门
func (d *Department) Deactivate() {
	d.Status = DepartmentStatusInactive
}

// Activate 激活部门
func (d *Department) Activate() {
	d.Status = DepartmentStatusActive
}

// DepartmentRepository 部门仓储接口
type DepartmentRepository interface {
	Create(department *Department) error
	Update(department *Department) error
	FindByID(id uint) (*Department, error)
	FindByCode(code string) (*Department, error)
	List(status *string, page, pageSize int) ([]*Department, int64, error)
	Delete(id uint) error
}
